docs/data/pages.ts

```diff
@@ -103,41 +103,63 @@ const pages: MuiPage[] = [
         title: 'API Reference',
         children: [
           { pathname: '/x/api/data-grid', title: 'Index' },
-          { pathname: '/x/api/data-grid/data-grid', title: 'DataGrid' },
-          { pathname: '/x/api/data-grid/data-grid-pro', title: 'DataGridPro' },
-          { pathname: '/x/api/data-grid/data-grid-premium', title: 'DataGridPremium' },
-          { pathname: '/x/api/data-grid/grid-api', title: 'GridApi' },
-          { pathname: '/x/api/data-grid/grid-cell-params', title: 'GridCellParams' },
-          { pathname: '/x/api/data-grid/grid-col-def', title: 'GridColDef' },
           {
-            pathname: '/x/api/data-grid/grid-single-select-col-def',
-            title: 'GridSingleSelectColDef',
-          },
-          { pathname: '/x/api/data-grid/grid-actions-col-def', title: 'GridActionsColDef' },
-          { pathname: '/x/api/data-grid/grid-export-state-params', title: 'GridExportStateParams' },
-          { pathname: '/x/api/data-grid/grid-filter-form', title: 'GridFilterForm' },
-          { pathname: '/x/api/data-grid/grid-filter-item', title: 'GridFilterItem' },
-          { pathname: '/x/api/data-grid/grid-filter-model', title: 'GridFilterModel' },
-          { pathname: '/x/api/data-grid/grid-filter-operator', title: 'GridFilterOperator' },
-          { pathname: '/x/api/data-grid/grid-filter-panel', title: 'GridFilterPanel' },
-          {
-            pathname: '/x/api/data-grid/grid-row-class-name-params',
-            title: 'GridRowClassNameParams',
-          },
-          { pathname: '/x/api/data-grid/grid-row-params', title: 'GridRowParams' },
-          { pathname: '/x/api/data-grid/grid-row-spacing-params', title: 'GridRowSpacingParams' },
-          {
-            pathname: '/x/api/data-grid/grid-aggregation-function',
-            title: 'GridAggregationFunction',
-          },
-          { pathname: '/x/api/data-grid/grid-csv-export-options', title: 'GridCsvExportOptions' },
-          {
-            pathname: '/x/api/data-grid/grid-print-export-options',
-            title: 'GridPrintExportOptions',
-          },
-          {
-            pathname: '/x/api/data-grid/grid-excel-export-options',
-            title: 'GridExcelExportOptions',
+            pathname: '/x/api/data-grid-components-group',
```

Comment:
I've grouped components and interfaces to make it cleaner:

<img width="269" alt="Screenshot 2023-02-24 at 00 27 10" src="https://user-images.githubusercontent.com/13808724/221054491-c642de30-30f8-4284-9b40-8f5b9c375855.png">